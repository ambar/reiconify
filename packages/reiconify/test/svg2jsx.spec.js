const {svg2jsx} = require('..')

let svg = `
<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
  <!-- comment -->
  <desc>desc</desc>
  <metadata>meta</metadata>
  <path /><!-- empty -->
  <path pointer-events="none" fill-opacity="0.5" d="M0 0h24v24H0z" /><!-- sort -->
</svg>
`

describe('svg2jsx', () => {
  it('converts svg to jsx', async () => {
    expect(await svg2jsx(svg)).toMatchSnapshot()
  })

  it('converts svg to jsx (React <= 15)', async () => {
    expect(await svg2jsx(svg, {camelCaseProps: true})).toMatchSnapshot()
  })

  it('uses `removeAttrs` plugin', async () => {
    const svgoPlugins = [
      {
        removeAttrs: {
          attrs: ['svg:(width|height|viewBox)'],
        },
      },
    ]
    expect(await svg2jsx(svg, {svgoPlugins})).toMatchSnapshot()
  })

  it('reuses svgo instance', async () => {
    const toJsx = svg2jsx.createSvg2jsx({
      camelCaseProps: true,
      svgoPlugins: [
        {
          removeAttrs: {attrs: 'fill-opacity'},
        },
      ],
    })
    expect(await toJsx(svg)).toMatchSnapshot()
  })
})
