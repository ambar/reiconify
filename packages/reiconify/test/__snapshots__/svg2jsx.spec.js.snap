// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`svg2jsx converts svg to jsx (React <= 15) 1`] = `"<svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\"><path d=\\"M0 0h24v24H0z\\" fillOpacity=\\".5\\" pointerEvents=\\"none\\"/></svg>"`;

exports[`svg2jsx converts svg to jsx 1`] = `"<svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\"><path fill-opacity=\\".5\\" d=\\"M0 0h24v24H0z\\" pointer-events=\\"none\\"/></svg>"`;

exports[`svg2jsx reuses svgo instance 1`] = `"<svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\"><path d=\\"M0 0h24v24H0z\\" pointerEvents=\\"none\\"/></svg>"`;

exports[`svg2jsx uses \`removeAttrs\` plugin 1`] = `"<svg><path fill-opacity=\\".5\\" d=\\"M0 0h24v24H0z\\" pointer-events=\\"none\\"/></svg>"`;
